### DISCUSSION TITLE
press+hold key, for temporary mouse pointer

### DISCUSSION THREAD

**Discussion by lucacuccu3x8** [pubpub-zz/ppInk#188] [Ideas]
@pubpub-zz 
Again, thank you so much for this amazing project.
I wondered if it was possible to add a press+hold button for the mouse pointer mode.
- While you'll "press+hold" the assigned button(or hotkey combo), you'll get the mouse pointer (with clicks and wheel)
- When you release and unhold the hotkey, you'll then get back to drawing mode.

Reason why: 
Sometimes, during lectures and live presentations, I have to scroll down pages or click on links quickly to go back into drawing mode either by static hotkey or clicking on the toolbar.
When you do this "back and forward" quickly, it tends to become confusing and tiresome at some point while speaking and presenting. 

Do you think this could be a nice and possible feature that could be implemented please?

**Comment by pubpub-zz** [comment]
Hi @lucacuccu3x8 ,
Thanks for the compliments. 
The only easy solution is to monitor the alt/shift/ctrl keys. in the attached zip, as default Ctrl+Alt+G will engage pointer, with the new option  (available on Hotkey tab), the pointer mode will release when you will release all keys (bot ctrl and Alt).
[ppInk02a.zip](https://github.com/pubpub-zz/ppInk/files/8819111/ppInk02a.zip)
also remember since #177 implementation,the global hotkey can be a simple key (e.g F1) - caution do not set the new option with such setup. Another option you could be interested in is to engage ppInk window mode and leave a part of the window below uncovered. If the mouse is out of ppink window, the mouse scroll will be processed by the below window.

give me your feedback and if you want to update some translation, copy and adjust the new line 
`OptionsExitPointerOnGlobalRelease = "Exit Pointer on all key release"`

**Reply by lucacuccu3x8** [reply]
Hi @pubpub-zz
Thank you for the prompt response.
I'm not sure if it works for me. 
https://user-images.githubusercontent.com/9900115/171604453-e084b3d3-a10e-450d-bec3-def6c4da92c5.mp4

However, I thought more on an "Exit drawing on full key hold" instead.
Here shown in this clip, I assigned the F8 for a test. 
When F8 is on hold, it keeps flickering (going back and forward from drawer to mouse) and sometimes the application crash.

**Reply by pubpub-zz** [reply]
You are on my caution/not working caseIf you set a single key, you will get multiple key 'WM_PRESS'-like events : this is why I can not intercept. 
Set Ctrl+f8 (for example) pressing shortly f8 and keeping Ctrl down. Do your stuff and release Ctrl when done. Pointer will be exited

**Reply by pubpub-zz** [reply]
@lucacuccu3x8  
+1?

**Comment by pubpub-zz** [comment]
@lucacuccu3x8 ,
   Any update?

**Reply by pubpub-zz** [reply]
without feed back, and due to material/time issue, the commit in ppink#2a has  not issued and integrated in 1.8.3
@lucacuccu3x8 still waiting for your update

**Comment by Wogrim** [comment]
I was also looking for a more convenient way to go in and out of mouse pointer mode.
I expect going into draw mode to remain a complicated hotkey to not conflict with other hotkeys in my software.
Currently using the hotkey on a Stream Deck button, although a keyboard or mouse with macro buttons would probably be better. 
Anyways, I was thinking going from draw mode to mouse pointer could be a separate, simpler hotkey like TAB.

**Reply by pubpub-zz** [reply]
@Wogrim,
the existing version allows already to use Tab (as key-press event) to enter and exit pointer , although this is not recommended as this key will be also processed by the active window (when pointer is engaged).
Note : As indicated above in the thread stream before the current beta about key modifiers release to exit pointer mode has not been released yet) but v1.8.3 should be good for you if you really want to use tab.
If you are using a Stream Deck, I would more likely recommend you to use the REST API to have a more tunable solution (you can activate pointer mode using http://localhost:7999/PenDef?P=-2 and return to hand mode using http://localhost:7999/PenDef?P=0)
PS : do not hesitate to star ppInk if you like it 😊
